pkg/controllers/management/rkeworkerupgrader/node.go

```diff
@@ -162,12 +162,21 @@ func (uh *upgradeHandler) filterNodes(nodes []*v3.Node, expectedVersion int) (ma
 	return toPrepareMap, toProcessMap, upgradedMap, notReadyMap, filtered, upgrading, done
 }
 
-func preparingNode(node *v3.Node) bool {
-	return node.Spec.DesiredNodeUnschedulable == "drain" || node.Spec.DesiredNodeUnschedulable == "true"
+func preparingNode(node *v3.Node, drain bool) bool {
+	if drain {
+		return node.Spec.DesiredNodeUnschedulable == "drain"
+	}
+	return node.Spec.DesiredNodeUnschedulable == "true"
 }
 
-func preparedNode(node *v3.Node) bool {
-	return v3.NodeConditionDrained.IsTrue(node) || node.Spec.InternalNodeSpec.Unschedulable || v3.NodeConditionUpgraded.IsUnknown(node)
+func preparedNode(node *v3.Node, drain bool) bool {
+	if v3.NodeConditionUpgraded.IsUnknown(node) {
+		return true
+	}
+	if drain {
```

Comment:
should we do this check first before `if v3.NodeConditionUpgraded.IsUnknown(node)`?